workflows/dfe.snake

```diff
@@ -0,0 +1,326 @@
+
+rule run_dfe_alpha:
+    input:
+        rules.generate_dfe_alpha_fs.output
+    output:
+        output_dir + "/inference/{demog}/DFE-alpha/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/nonneu/est_dfe.out"
+    threads: 1
+    resources: time_min=60, mem_mb=5000, cpus=1
+    shell:
+        "est_dfe -c {input[0]}; est_dfe -c {input[1]}"
+
+rule get_dfe_alpha_bestfits:
+    input:
+        rules.run_dfe_alpha.output
+    output:
+        output_dir + "/inference/{demog}/DFE-alpha/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.DFE-alpha.bestfit"
+    threads: 1
+    resources: time_min=60, mem_mb=5000, cpus=1
+    shell:
+        """
+        cat {input} | awk -v id={wildcards.ids} '{{print "pop"id"\t"$22"\t"$10"\t"$12}}' | sed '1ipop_id\tlikelihood\tb\tEs' > {output}
+        """
+
+# ###############################################################################
+# grapes
+# ###############################################################################
+
+# ###############################################################################
+# plots
+# ###############################################################################
+
+
+rule plot_results:
+    input:
+        dadi_res = expand(output_dir + "/inference/{demog}/dadi/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.dadi.bestfit", 
+            demog=demo_model_id_list,
+            seeds=seed_list,
+            dfes=dfe_list,
+            annots=annotation_list,
+            chrms=chrm_list, 
+            ids=[0, 1, 2]
+        ),
+        polydfe_res = expand(output_dir + "/inference/{demog}/polyDFE/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.polyDFE.bestfit",
+            demog=demo_model_id_list,
+            seeds=seed_list,
+            dfes=dfe_list,
+            annots=annotation_list,
+            chrms=chrm_list,
+            ids=[0,1,2]
+        ),
+        dfe_alpha_res = expand(output_dir + "/inference/{demog}/DFE-alpha/{dfes}/{annots}/{seeds}/{chrms}/pop{ids}/pop{ids}.DFE-alpha.bestfit",
+            demog=demo_model_id_list,
+            seeds=seed_list,
+            dfes=dfe_list,
+            annots=annotation_list,
+            chrms=chrm_list,
+            ids=[0,1,2]
+        )
+    output:
+        output_dir + "/plots/{demog}/{chrms}/{dfes}/{annots}/dfe.inference.benchmark.pdf"
+    threads: 1
+    resources: time_min=60, mem_mb=5000, cpus=1
+    run:
+        contig = species.get_contig(wildcards.chrms)
+        seq_len = contig.recombination_map.sequence_length
+
+        dadi_bestfits = [ b for b in input.dadi_res if wildcards.demog in b]
+        polydfe_bestfits = [ b for b in input.polydfe_res if wildcards.demog in b]
+        dfe_alpha_bestfits = [ b for b in input.dfe_alpha_res if wildcards.demog in b]
```

Comment:
@andrewkern I separated input files by demographic models.

Reply:
looks good. i don't know what these files are though-- are they auto written by the code?

Reply:
`input.dadi_res`, `input.polydfe_res`, and `inpu.dfe_alpha_res` collect all results regardless of demographic models. Therefore, the final figures for the `Constant` and `OutOfAfrica_3G09` models are the same. I need to extract results based on demographic models and make plots.